### PR TITLE
Makes sure sorting orders by execution order is deterministic.

### LINKED ISSUE #75: Could not round solution (nr_exec_orders > max_nr_exec_orders)
The open solver had an issue with rounding a solution for this instance:
https://gnosis-dev-dfusion.s3.amazonaws.com/data/mainnet_dev/open-solver/instances/2020-08-03/instance_5321545_2020-08-03T14%3A10%3A31.561627002%2B00%3A00.json

 File "/app/open_solver/src/core/validation.py", line 41, in validate
   assert nr_exec_orders <= max_nr_exec_orders
AssertionError
(see https://gnosisinc.slack.com/archives/CQZBU9T5J/p1596463904003100)

It happened again in the subsequent batch:
https://gnosis-dev-dfusion.s3.amazonaws.com/data/mainnet_dev/open-solver/instances/2020-08-03/instance_5321546_2020-08-03T14%3A15%3A31.552265513%2B00%3A00.json

### PR DESCRIPTION
**Pull request by marcovc**
Sorting orders by execution priority was not stable (e.g. orders with the same max_xrate would be sorted non-deterministically). This is true even if python default sorting algorithm is stable, since lists of orders to sort are sometimes shortened.
Closes #75

### CODE REVIEW

File: dex_open_solver/token_pair_solver/solver.py [hunk 1]
```diff
@@ -359,6 +358,7 @@ def solve_token_pair_and_fee_token(
             # In other words, we can stop trying to augmenting the set of f_orders
             # once adding a new f_order degrades the objective.
             if best_objective is not None and objective < best_objective:
+                print("leaving here")
```
Comment (josojo) [inline dex_open_solver/token_pair_solver/solver.py:361]: did you forget to remove it?

File: dex_open_solver/token_pair_solver/solver.py [hunk 2]
```diff
@@ -369,6 +369,7 @@ def solve_token_pair_and_fee_token(
             )
 
         xrate, b_buy_token_price, b_orders, s_orders, f_orders = best_solution
+        print(len(b_orders), len(s_orders), len(f_orders))
```
Comment (josojo) [inline dex_open_solver/token_pair_solver/solver.py:372]: same?

**Review by josojo — APPROVED** [review]
Tricky issue.

Thanks a lot for debugging and finding that!

File: dex_open_solver/token_pair_solver/solver.py (hunk 1 above)
Comment (marcovc) [reply dex_open_solver/token_pair_solver/solver.py]: oops, thank you!

**Pull request merged**